src/background.ts

```diff
@@ -122,17 +124,46 @@ function getQr(
   });
 }
 
-async function getTotp(text: string) {
+async function getTotp(text: string, silent = false) {
   if (!contentTab || !contentTab.id || !text) {
-    return;
+    return false;
   }
   const id = contentTab.id;
 
   if (text.indexOf("otpauth://") !== 0) {
-    if (text === "error decoding QR Code") {
-      chrome.tabs.sendMessage(id, { action: "errorqr" });
+    if (text.indexOf("otpauth-migration://") === 0) {
+      const otpUrls = getOTPAuthPerLineFromOPTAuthMigration(text);
+      if (otpUrls.length === 0) {
+        !silent && chrome.tabs.sendMessage(id, { action: "errorenc" });
+        return false;
+      }
+
+      let failedCount = 0;
+      for (const otpUrl of otpUrls) {
+        const success = await getTotp(otpUrl, true);
```

Comment:
I noticed that it hangs when importing a lot of codes, mabye promises.allSettled these right after the loop?

Reply:
Good suggestion